### PR TITLE
Fix `TensorIterator::view_offsets_` size

### LINKED ISSUE #37084: argmax and argmin ops return the random number for scalar tensor when specifying dim
## 🐛 Bug

argmax and argmin ops return the random number for scalar tensor when specifying dim
 
## To Reproduce
```
>>> import torch
>>> torch.__version__
'1.5.0'
>>> a = torch.tensor(5.)
>>> torch.argmax(a) # it gives the right result
tensor(0)
>>> torch.argmax(a, dim=0)
tensor(140457834750256)
```

cc @ezyang @gchanan @zou3519

### PR DESCRIPTION
**Pull request by emcastillo**
Fixes #37084

There are 3 alternatives for this design.

This PR and the first one.
When a tensor is a scalar `ndim==0`, accessing view_offsets_[0] when doing reductions, yields an invalid offset for the index which is the output of `argmax` and `argmin`. 

https://github.com/pytorch/pytorch/blob/fba9b9a023a107c9ad33bfcc834146e362616b19/aten/src/ATen/native/cpu/Reduce.h#L217

This also happens in cuda code:
https://github.com/pytorch/pytorch/blob/fba9b9a023a107c9ad33bfcc834146e362616b19/aten/src/ATen/native/cuda/Reduce.cuh#L797

The second alternative is to check the size of `view_offsets` before accessing it. But this introduces some burden.

The third alternative is related to the way that inputs are treated in `argmax` and `argmin`
depending on the `dim` argument value.

https://github.com/pytorch/pytorch/blob/fba9b9a023a107c9ad33bfcc834146e362616b19/aten/src/ATen/native/ReduceOps.cpp#L775-L780

If `dim` is not specified, then the scalar gets reshaped into a 1-dim tensor and everything works properly, since now `view_offsets` has an actual entry.
If dim is specified, then the input remains as a scalar causing the issue we see here.

This PR tries to solve it in a generic way for every case so I went with option 1. I am willing to discuss it and change if you think that the other alternatives are better.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -8585,6 +8585,32 @@ def run_test(A_dims, b_dims, pivot=True):
         run_test((4, 4), (2, 1, 3, 4, 2))  # broadcasting A
         run_test((1, 3, 1, 4, 4), (2, 1, 3, 4, 5))  # broadcasting A & b
 
+    def test_dim_arg_reduction_scalar(self, device):
+        example = 4.0
+
+        types = [torch.double,
```
Comment (ngimel) [inline test/test_torch.py:8591]: You can use @dtypes, @dtypesIfCUDA decorators.

**Review by ngimel — APPROVED** [review]
Looks good, thanks for the fix, please use dtype decorator in the fix.

File: test/test_torch.py (hunk 1 above)
Comment (emcastillo) [reply test/test_torch.py]: Done, thanks!

**Comment by emcastillo** [comment]
PTAL :)
